### PR TITLE
chore(typings): Remove typings - Add @types

### PR DESCRIPTION
**Pull request by MarkPieszak**
Moved typed-xhr2 to custom_typings folder to remove typings completely.

Upgrade to Typescript@beta (2)

Closes #479 #478

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,9 +100,16 @@
     "rxjs": "5.0.0-beta.6",
     "through2": "^2.0.0",
     "ts-node": "^0.9.3",
-    "typescript": "^2.0.0-dev.20160706",
-    "typings": "^1.0.5",
+    "typescript": "^2.0.0",
     "xhr2": "^0.1.3",
     "zone.js": "~0.6.12"
+  },
+  "dependencies": {
```
Comment (gdi2290) [inline package.json]: can you keep these in devDeps

File: tsconfig.json [hunk 2]
```diff
@@ -23,10 +23,21 @@
...
+      "core-js",
+      "es6-promise",
+      "jasmine",
+      "node",
+      "through2",
+      "zone.js"
+    ]
   },
 
   "files": [
```
Comment (gdi2290) [inline tsconfig.json]: there should be glob support here now. can you check

File: types.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/// <reference path="node_modules/@types/browserify/index.d.ts" />
```
Comment (MarkPieszak) [inline types.d.ts]: I'm not sure, was talking to Jeff about this he said VSCode has some issues with these things. I have the latest 1.3.1. I can remove this file if you don't think we need it then. @gdi2290

File: tsconfig.json (hunk 2 above)
Comment (MarkPieszak) [inline tsconfig.json]: I don't think these types are necessary they just help the compiler? Should I remove them for now?
Not sure on globs impossible to find any good info on types at the moment.

Just saw a post from @DanielRosenwasser ([here](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/#comment-19085))
The compiler will automatically find global declarations from the first @types folder, and if you you need the compiler to search a little farther, there will be a ‘types’ property in ‘package.json’ to specify that you need a specific set of files, which would look something like this:
```
{
"compilerOptions": {
"types": ["jasmine", "jquery"]
}
```
Think he meant tsconfig, (types option shows up there)
image

File: package.json (hunk 1 above)
Comment (MarkPieszak) [inline package.json]: Done

File: types.d.ts (hunk 3 above)
Comment (gdi2290) [inline types.d.ts]: let's keep it

File: package.json [hunk 4]
```diff
@@ -70,6 +70,13 @@
     "@angular/platform-server": "2.0.0-rc.4",
     "@angular/router": "3.0.0-alpha.8",
     "@angular/router-deprecated": "2.0.0-rc.2",
+    "@types/jasmine": "^2.2.28-alpha",
+    "@types/browserify": "^12.0.28-alpha",
+    "@types/core-js": "^0.9.27-alpha",
+    "@types/es6-promise": "0.0.27-alpha",
```
Comment (gdi2290) [inline package.json]: we have core-js so we can remove es6-promise

Comment (gdi2290) [inline package.json]: if we're not including core-js anywhere we can stick with es6-promise or use the `"lib"` prop in tsconfig.

**Comment by gdi2290** [comment]
can you also create index.d.ts with
```
declare module '*';
```

**Comment by gdi2290** [comment]
@MarkPieszak add declare module '*'; then it LGTM

**Comment by MarkPieszak** [comment]
Is it okay that it throws a ts error `'{' is expected` ?

**Comment by gdi2290** [comment]
LGTM if `declare module '*';` works

**Pull request merged**